docs/core-components.md

```diff
@@ -0,0 +1,201 @@
+
+| Name         | Type        | Description                                                                                                            | Default |
+|--------------|-------------|------------------------------------------------------------------------------------------------------------------------|---------|
+| `model.open` | `{Boolean}` | Whether our associated [MIDIPort](https://developer.mozilla.org/en-US/docs/Web/API/MIDIPort) should be open or closed. | `false` |
+
+### Events
+
+| Event         | Description                                               |
+|---------------|-----------------------------------------------------------|
+| `onPortOpen`  | Fired when our port is opened successfully.               |
+| `onPortClose` | Fired when our port is closed successfully.               |
+| `onError`     | Fired when there is an error opening or closing our port. |
+
+## `youme.connection.input`
+
+A connection to a [MIDIInput](https://developer.mozilla.org/en-US/docs/Web/API/MIDIInput).
+
+### Events
+
+In addition to the events for the base `youme.connection` grade, this grade has the following events.
+
+| Event           | Description                                                         |
+|-----------------|---------------------------------------------------------------------|
+| `onActiveSense` | An event that is fired when an "active sense" message is received.  |
+| `onAftertouch`  | An event that is fired when an "after touch" message is received.   |
+| `onClock`       | An event that is fired when a "clock" message is received.          |
+| `onContinue`    | An event that is fired when a "clock continue" message is received. |
+| `onControl`     | An event that is fired when an "control" message is received.       |
+| `onMessage`     | An event that is fired when any message is received.                |
+| `onNoteOn`      | An event that is fired when a "note on" message is received.        |
+| `onNoteOff`     | An event that is fired when a "note off" message is received.       |
+| `onPitchbend`   | An event that is fired when a "pitch bend" message is received.     |
+| `onProgram`     | An event that is fired when a "program change" message is received. |
+| `onReset`       | An event that is fired when a "reset" message is received.          |
+| `onSongPointer` | An event that is fired when a "song pointer" message is received.   |
+| `onSongSelect`  | An event that is fired when a "song select" message is received.    |
+| `onStart`       | An event that is fired when a "clock start" message is received.    |
+| `onStop`        | An event that is fired when a "clock stop" message is received.     |
+| `onSysex`       | An event that is fired when a "sysex" message is received.          |
+| `onTuneRequest` | An event that is fired when a "tune request" message is received.   |
+
+See [the WebMIDI API documentation](https://www.midi.org/specifications-old/item/table-1-summary-of-midi-message) for
```

Comment:
This looks like it links to the MIDI specification, rather than the WebMIDI API documentation as the link text suggests.

Reply:
Yes, the WebMIDI API links to that rather than spell it out on its own, so I changed it to "the MIDI specificiation".